### ISSUE TITLE
[Question] vLLM Hidden States Extraction

### ISSUE THREAD

**Issue by shadowpa0327** [torchspec-project/TorchSpec#87]
Thanks for the awesome work and open-sourcing these amazing projects! I have some questions on using vLLM's to collect hidden states online.

Would the captured last hidden states for training **off by 1**? 
When I check the initialization for `VllmEngine`, I add some debug prints to verify the arguments we specified for `eagle_aux_hidden_state_layer_ids`. As shown from the screenshot below (Qwen3-8B with 36 layers), the debug message shows that the `35` is appended to collect the last hidden states for computing logits.

<img width="1308" height="58" alt="Image" src="https://github.com/user-attachments/assets/c39487e1-675a-4348-a630-a3200115f9d3" />

But given that the vLLM captures the hidden states using the snippets below

```python
aux_hidden_states = self._maybe_add_hidden_state([], 0, hidden_states, residual)
for idx, layer in enumerate(islice(self.layers, self.start_layer, self.end_layer)):
    hidden_states, residual = layer(positions, hidden_states, residual)
    self._maybe_add_hidden_state(
        aux_hidden_states, idx + 1, hidden_states, residual
    )
```

It seems we need to set the last layer idx to 36 instead of 35 to correctly get the `hidden_states` before `norm` & `lm_head`? 

I manually made the change and printed the hidden states; setting layer idx to 36 might be a more correct and expected behavior? 

<img width="1246" height="184" alt="Image" src="https://github.com/user-attachments/assets/5b6ae3ce-0696-4d45-9f10-4ef917c16cda" />

There might be some implementation details I missed. I hope for clarification! Thanks in advance!

Sincerely,
Chi-Chih

**Comment by yubofredwang** [comment]
hi @shadowpa0327 thanks a lot for reporting this issue. yes you are right, the layer idx should be 36. I already created a fix in the main. 

Also contributions are welcomed! Feel free to create a fix as well for future issues.
